src/main/java/com/google/sps/data/NaturalLanguageProcessor.java

```diff
@@ -0,0 +1,155 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import com.google.cloud.language.v1.AnalyzeEntitiesRequest;
+import com.google.cloud.language.v1.AnalyzeEntitiesResponse;
+import com.google.cloud.language.v1.ClassifyTextRequest;
+import com.google.cloud.language.v1.ClassifyTextResponse;
+import com.google.cloud.language.v1.ClassificationCategory;
+import com.google.cloud.language.v1.Document;
+import com.google.cloud.language.v1.Document.Type;
+import com.google.cloud.language.v1.EncodingType;
+import com.google.cloud.language.v1.Entity;
+import com.google.cloud.language.v1.EntityMention;
+import com.google.cloud.language.v1.LanguageServiceClient;
+import java.util.Set;
+import java.util.List;
+import java.util.HashSet;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.lang.Exception;
+
+/** Picks entities from text using Google's NLP API */
+public class NaturalLanguageProcessor {
+
+    private static final double DEFAULT_SALIENCE_THRESHOLD = 0.01;
+    private static final double ACADEMIC_SALIENCE_THRESHOLD = 0.02;
+    private static final double CATEGORY_THRESHOLD = 0.7;
+    private static final String WORD_DELIMITER = " ";
+    private static final String CATEGORY_DELIMITER = "/";
+    private static final int MIN_TOKENS_FOR_CATEGORY_CLASSIFICATION = 20;
+    private static final int CATEGORY_INDEX = 1;
+    private static final String[] ACADEMIC_CATEGORIES = {
```

Comment:
Are we planning on actually using Academic Categories? What for? Organization if we decide to display all videos?

Reply:
These are the categories used by the NLP API, and are used to determine if the context is academic, and not currently used for anything on the front end or database parts.